File: server/php/src/Application/Domain/Payments/PaymentDetails/ReattemptInstructionsConditions.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class ReattemptInstructionsConditions
{
    public ?int $maxAttempts = null;
    public ?int $maxDelay = null;

    public function toArray(): array
    {
        return [
            'maxAttempts' => $this->maxAttempts,
            'maxDelay' => $this->maxDelay,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/reattempt_instructions_conditions.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class ReattemptInstructionsConditions
          attr_accessor :max_attempts, :max_delay

          def initialize(max_attempts: nil, max_delay: nil)
            @max_attempts = max_attempts
            @max_delay = max_delay
          end

          def to_h
            {
              'maxAttempts' => @max_attempts,
              'maxDelay' => @max_delay
            }
          end
        end
      end
    end
  end
end